### PR TITLE
Add Python 3.8; drop Python <3.6

### LINKED ISSUE #133: Python 3.8 DeprecationWarnings; drop support for legacy Python
On Python 3.8, any application using asyncio_redis inundates the log with:

> lib/python3.8/site-packages/asyncio_redis/protocol.py:805: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    self._transaction_lock = asyncio.Lock(loop=loop)

> lib/python3.8/site-packages/asyncio_redis/protocol.py:831: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def initialize():

I don't think the problem is solvable without dropping support for Python < 3.5.
I would love to see asyncio-redis adopt NEP-29 <https://numpy.org/neps/nep-0029-deprecation_policy.html>, which would mean dropping support for Python < 3.6.

We've already pushed NEP-29 to the xarray and pint projects, with great satisfaction of all developers and zero complaints from the users.

### PR DESCRIPTION
**Pull request by crusaderky**
* Closes #133
* Added support for Python 3.8
* Dropped support for Python <3.6
* Deprecated 'loop' parameter in Connection.create and Pool.create
* Fixed all DeprecationWarnings
* Fix missing object asyncio.streams.IncompleteReadError
* Fixed all Sphinx build errors and warnings
* Intersphinx: all classes that can be imported from the top-level module must now be
  referenced from the top-level module in the docstrings too.
  E.g. ``:class:`asyncio_redis.exceptions.Error` `` has changed to
 ``:class:`asyncio_redis.Error` ``.

### CODE REVIEW

**Comment by crusaderky** [comment]
@jonathanslenders ready for review

**Comment by crusaderky** [comment]
Hi @jonathanslenders, any feedback on this?

**Comment by jonathanslenders** [comment]
Hi @crusaderky,

Thank you so much for this! It looks great.
If you think it's ready and well tested, I can merge it.

**Comment by crusaderky** [comment]
@jonathanslenders I tested it in my Python 3.8 project, which however uses only a tiny subset of the methods, and it seems to work fine. For everything else, we have to rely on the unit tests. I'd say merge...

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,20 @@
 CHANGELOG
 =========
 
+0.16.0: unreleased
+------------------
+- Added support for Python 3.8
+- Dropped support for Python <3.6
+- Deprecated 'loop' parameter in Connection.create and Pool.create
+- Added than support for zpopmin and zrangebylex methods
```
Comment (crusaderky) [inline CHANGELOG]:
```suggestion
- Added support for zpopmin and zrangebylex methods
```

File: CHANGELOG [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
+------------------
+- Added support for Python 3.8
+- Dropped support for Python <3.6
+- Deprecated 'loop' parameter in Connection.create and Pool.create
+- Added support for zpopmin and zrangebylex methods
+- Fix DeprecationWarning for inspect.formatargspec()
+- Fix missing object asyncio.streams.IncompleteReadError
+- Intersphinx: all classes that can be imported from the top-level module must now be
+  referenced from the top-level module in the docstrings too.
+  E.g. :class:`asyncio_redis.exceptions.Error` has changed to
```
Comment (crusaderky) [inline CHANGELOG]:
```suggestion
  E.g. asyncio_redis.exceptions.Error has changed to
```

File: CHANGELOG [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
+  :class:`asyncio_redis.Error`.
```
Comment (crusaderky) [inline CHANGELOG]:
```suggestion
  asyncio_redis.Error.
```

**Comment by jonathanslenders** [comment]
Merged! Thanks a lot for this!

**Comment by jbasko** [comment]
This was never released, was it? Is the idea to use a version pointing to git commit?

**Pull request merged**